tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java

```diff
@@ -136,14 +135,17 @@ public void lineageCompleteAndAsyncPersistTest() throws Exception {
       HeartbeatScheduler.schedule(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING);
       Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING, 5,
           TimeUnit.SECONDS));
+      status = getFileSystemMasterClient().getStatus(uri);
       HeartbeatScheduler.schedule(HeartbeatContext.WORKER_FILESYSTEM_MASTER_SYNC);
       Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_FILESYSTEM_MASTER_SYNC, 5,
           TimeUnit.SECONDS));
-
-      status = getFileSystemMasterClient().getStatus(uri);
       Assert.assertEquals(PersistenceState.IN_PROGRESS.toString(), status.getPersistenceState());
+      CommonUtils.sleepMs(500);
```

Comment:
In theory yes, however, the operation on worker is in a thread shouldn't take more than 100ms. A better guarantee is to implement an augmented version of `wait()` method that heartbeats every 20ms.